base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -0,0 +1,112 @@
+import json
+import argparse
+import pathlib
+
+from packageurl import PackageURL
+
+
+def get_base_images_sbom_data(images, is_last_from_scratch):
+    """
+    Creates the base images sbom data
+
+    :param images: (List) - list of base images used in Dockerfile
+    :param is_last_from_scratch: (Boolean) - Is the last stage/base image from scratch?
+    :return: base_images_sbom_data (Dict) - formulation part of sbom, that contains base image components
+    """
+
+    base_images_sbom_data = {"formulation": [{"components": []}]}
+
+    # property_name shows whether the image was used only in the building process
+    # or if it is the final base image. If the final base image is scratch, then
+    # this is omitted, because we aren't including scratch in the sbom.
+    for index, image in enumerate(images):
+        property_name = f"konflux:container:is_builder_image:for_stage"
+        property_value = str(index)
+        if index == len(images) - 1 and not is_last_from_scratch:
+            property_name = f"konflux:container:is_base_image"
+            property_value = "true"
+
+        # example image: registry.access.redhat.com/ubi8/ubi:latest@sha256:627867e53ad6846afba2dfbf5cef1d54c868a9025633ef0afd546278d4654eac
+        # repository = registry.access.redhat.com/ubi8/ubi:latest
+        # digest = sha256:627867e53ad6846afba2dfbf5cef1d54c868a9025633ef0afd546278d4654eac
+        # repository_without_tag = registry.access.redhat.com/ubi8/ubi
+        # tag = latest
+        # name = ubi
+        repository, digest = image.split("@")
+        repository_without_tag, tag = repository.split(":")
+        # last fragment of the repository
+        name = repository_without_tag.split("/")[-1]
```

Comment:
Tag can be `<none>`

The registry part can be in form `host:port`. 

Tag can be searched from the right side of the string, like https://github.com/tkdchen/my_image_utils/blob/main/src/my_image_utils/__init__.py#L139, or extract it with a regex that matches `:[0-9a-zA-Z._-]+$`.

Images built by Konflux have multiple path components in repository name, for example:
`quay.io/redhat-user-workloads/rh-acs-tenant/acs/collector:d92c93ab73f81442d99870528442855c75b535d7`

Reply:
Ok, so the fix was rather simple and this allowed to keep the whole functionality simple.
I extracted the image parsing to a function and added unit tests for it, so it's clear to see how it works and to verify it's functionality.

But if you still see some potential issues or would like to extend the testing, please let me know